data/soapvalidator/Admin/PrimaryEmailAddressChange/ZCS-5359.xml

```diff
@@ -0,0 +1,659 @@
+	<t:property name="test_account8.password" value="${defaultpassword.value}" />
+	<t:property name="test_account10.name"
+		value="test10.${TIME}.${COUNTER}@${defaultdomain.name}" />
+	<t:property name="test_account10.newname"
+		value="test10.${TIME}.${COUNTER}@${defaultdomain.name}" />
+	<t:property name="test_account10.password" value="${defaultpassword.value}" />
+
+	<t:test_case testcaseid="Ping" type="always">
+		<t:objective>basic system check</t:objective>
+
+		<t:test required="true">
+			<t:request>
+				<PingRequest xmlns="urn:zimbraAdmin" />
+			</t:request>
+			<t:response>
+				<t:select path="//admin:PingResponse" />
+			</t:response>
+		</t:test>
+
+	</t:test_case>
+
+
+	<t:test_case testcaseid="admin_auth" type="always">
+		<t:objective>login as the admin</t:objective>
+
+		<t:test required="true">
+			<t:request>
+				<AuthRequest xmlns="urn:zimbraAdmin">
+					<name>${admin.user}</name>
+					<password>${admin.password}</password>
+				</AuthRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//admin:AuthResponse/admin:authToken" set="authToken" />
+			</t:response>
+		</t:test>
+
+	</t:test_case>
+
+	<t:test_case testcaseid="PrimaryEmailAddressChange_01"
+		type="smoke" bugids="ZCS-5359">
+		<t:objective>
+			ChangePrimaryEmail Address of an account by name.
+			<t:steps>
+				1. Create a new account.
+				2. Fire ChangePrimaryEmailRequest to change it's email address.
+				3. GetAccount request to verify both old and new account exists.
+			</t:steps>
+		</t:objective>
+
+		<t:test id="createAccountRequest01">
+			<t:request>
+				<CreateAccountRequest xmlns="urn:zimbraAdmin">
+					<name>${test_account1.name}</name>
+					<password>${test_account1.password}</password>
+				</CreateAccountRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//admin:CreateAccountResponse/admin:account"
+					attr="id" set="test_account1.id" />
+			</t:response>
+		</t:test>
+
+		<t:test id="changePrimaryEmailAddress01">
+			<t:request>
+				<ChangePrimaryEmailRequest xmlns="urn:zimbraAdmin">
+					<account by="name">${test_account1.name}</account>
+					<newName>${test_account1.newname}</newName>
+				</ChangePrimaryEmailRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//admin:ChangePrimaryEmailResponse" />
+			</t:response>
+		</t:test>
+
+		<!-- Get account for old email address. It should pass-->
+		<t:test>
+			<t:request>
+				<GetAccountRequest xmlns="urn:zimbraAdmin">
+					<account by="name">${test_account1.name}</account>
+				</GetAccountRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//admin:GetAccountResponse/admin:account" />
+			</t:response>
+		</t:test>
+
+		<!-- Get account for new email address. It should pass -->
+		<t:test>
+			<t:request>
+				<GetAccountRequest xmlns="urn:zimbraAdmin">
+					<account by="name">${test_account1.newname}</account>
+				</GetAccountRequest>
+			</t:request>
+			<t:response>
+				<t:select path="//admin:GetAccountResponse/admin:account" />
```

Comment:
add verification that zimbraOldMailAddress is empty and zimbraPrimaryMailChangeHistory is populated (can add check that it contains the old mail address)